cfme/tests/networks/test_components_crud.py

```diff
@@ -0,0 +1,232 @@
+import pytest
+import fauxfactory
+from widgetastic.exceptions import NoSuchElementException
+from widgetastic_patternfly import DropdownItemDisabled
+
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.networks.cloud_network import CloudNetworkCollection
+from cfme.networks.floating_ip import FloatingIPCollection
+from cfme.networks.network_router import NetworkRouterCollection
+from cfme.networks.security_group import SecurityGroupCollection
+from cfme.networks.subnet import SubnetCollection
+from cfme.utils import testgen, version
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.wait import wait_for
+
+
+pytest_generate_tests = testgen.generate(classes=[OpenStackProvider], scope='function')
+pytestmark = pytest.mark.usefixtures('setup_provider')
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.8')
+@pytest.mark.parametrize("ext_gateway_state", [True, False])
+@pytest.mark.parametrize("source_nat", [True, False])
+def test_adding_router(provider, appliance, ext_gateway_state, source_nat):
+    """ Test crud of sdn router """
+    test_name = "test_router_" + fauxfactory.gen_alphanumeric(6)
+    network_manager = provider.name + " Network Manager"
+
+    collection = NetworkRouterCollection(appliance)
+    try:
+        view = navigate_to(collection, 'AddNewNetworkRouter')
+    except DropdownItemDisabled:
+        raise DropdownItemDisabled("Current provider doen't support adding Network Router")
+
+    view.name_field.fill(test_name)
+    view.provider_dropdown.select_by_visible_text(network_manager)
+
+    view.external_gateway.fill(ext_gateway_state)
```

Comment:
Here and below. I don't understand why do you create a network router via view? There should create method.